javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -73,71 +76,91 @@ private static int treeSize(int branchCount, int depthShift) {
         }
     }
 
-    BitMappedTrie<T> prepend(T leading) {
-        final int newSize = length() + 1;
-        if (length() == 0) {
-            return new BitMappedTrie<>(asArray(leading), offset, newSize, depthShift);
-        } else {
-            Object[] array = this.array;
-            int shift = depthShift, offset = this.offset;
-            if (isFullLeft()) {
+    BitMappedTrie<T> prependAll(Iterable<? extends T> iterable) {
```

Comment:
generalized `prepend` to reuse all newly inserted arrays (minimizing the copies)